### PR TITLE
Made Improvments On fda_testing.py, oct.py, fda.py

### PR DESCRIPTION
**Pull request by FurkyT**
- There is a warning when writing video from bscans says:
"Lossy conversion from float64 to uint8. Range [0.0, 255.0]. Convert image to uint8 prior to saving to suppress this warning."

- When testing the files with fda_testing.py if an image chunk is not in fda file methods raise Value Error and
the code terminating for preventing that the if condition in image methods updated as other methods. Like if an
image chunk is not in fda file the method print the info and return None. But this cause another error which is terminate
the code on testing side the error come from save function and says "None type object has no attribute save" for
prevent that an if statement added on fda_testing.py checking if image data exist or None.

- I made the changes you want on the fda.py now there is only one method for making the json from other chunks data

### CODE REVIEW

File: oct_converter/readers/fda.py [hunk 1]
```diff
@@ -186,9 +198,17 @@ def __init__(self, filepath):
...
+        )
+
+        self.fast_q2_info_header = Struct("various_quality_statistics" / Float32n[6])
+
+        self.gla_littmann_01_header = Struct("0xffff" / Int32un, "0x1" / Int32un)
+
         self.chunk_dict = self.get_list_of_file_chunks()
 
-    def get_list_of_file_chunks(self):
+    def get_list_of_file_chunks(self, show=True):
```
Comment (marksgraham) [inline oct_converter/readers/fda.py:211]: Could we rename to print?

File: oct_converter/readers/fda.py [hunk 2]
```diff
@@ -224,7 +246,8 @@ def read_oct_volume(self):
         """
 
         if b"@IMG_JPEG" not in self.chunk_dict:
-            raise ValueError("Could not find OCT header @IMG_JPEG in chunk list")
+            print("@IMG_JPEG IS NOT IN CHUNKS SKIPPING...")
```
Comment (marksgraham) [inline oct_converter/readers/fda.py:249]: Being pedantic but could we change to `print("Could not find OCT header @IMG_JPEG in chunk list, skipping.")` I think the capitals make it look like we're shouting :P

File: oct_converter/readers/fda.py [hunk 3]
```diff
@@ -254,7 +277,8 @@ def read_oct_volume_2(self):
         """
 
         if b"@IMG_MOT_COMP_03" not in self.chunk_dict:
-            raise ValueError("Could not find OCT header @IMG_MOT_COMP_03 in chunk list")
+            print("@IMG_MOT_COMP_03 IS NOT IN CHUNKS SKIPPING...")
```
Comment (marksgraham) [inline oct_converter/readers/fda.py:280]: And similar changes for all these readers

File: examples/fda_testing.py [hunk 4]
```diff
@@ -7,62 +7,68 @@
...
+            b"@IMG_TRC_02",
+        ]:  # this chunks are image chunks and extract with another methods
+            continue
+        json_key = key.decode().split("@")[-1].lower()
+        try:
+            output_json[json_key] = fda.read_any_info_and_make_dict(key)
+        except KeyError:
+            print(f"{key} there is no method for getting info from this chunk.")
 
     info_json = json.dumps(output_json)
```
Comment (marksgraham) [inline examples/fda_testing.py:73]: the output json file is much more readable with `json.dumps(output_json, indent=4)`

File: oct_converter/readers/fda.py [hunk 5]
```diff
@@ -186,9 +198,17 @@ def __init__(self, filepath):
             "size" / Int32un,
         )
 
+        self.align_info_header = Struct(
```
Comment (marksgraham) [inline oct_converter/readers/fda.py:201]: we have started to move all the structs into their own files, under `readers/binary_structs` - there is an example for boct currently, and import them from there. I think doing this here will keep the FDA reader more readable and reduce the number of class attributes

**Review by marksgraham — CHANGES_REQUESTED** [review]
Looks pretty good! I've requested a few changes

File: oct_converter/readers/fda.py (hunk 1 above)
Comment (FurkyT) [reply oct_converter/readers/fda.py:211]: Yes no problem will do it now.

File: oct_converter/readers/fda.py (hunk 2 above)
Comment (FurkyT) [reply oct_converter/readers/fda.py:249]: Oh okay I did not think like that :D

File: examples/fda_testing.py (hunk 4 above)
Comment (FurkyT) [reply examples/fda_testing.py:73]: Okay adding this to dumps part.

File: oct_converter/readers/fda.py (hunk 5 above)
Comment (FurkyT) [reply oct_converter/readers/fda.py:201]: Okay I'll try to make files like that.

**Comment by FurkyT** [comment]
Hi,
Made the changes you want and added struct to the readers/binary_structs folder. 
Files and codes are tested with one of my private fda files.

**Review by marksgraham — APPROVED** [review]

**Comment by FurkyT** [comment]
Its been a pleasure for me.

**Pull request merged**